letta-ai/letta
# feat: Enable basic sandboxed tool run functionality

**Please describe the purpose of this pull request.**
Basic functionality to support running user-specified tools in sandbox. Submitting v1 for feedback on approach! 

**How to test**
How can we test your PR during review? What commands should we run? What outcomes should we expect?
```
poetry run pytest -s -vv tests/test_tools.py::test_run_basic_tool_in_sandbox
poetry run pytest -s -vv tests/test_tools.py::test_run_tool_with_params_in_sandbox
```

**Have you tested this PR?**
Have you tested the latest commit on the PR? If so please provide outputs from your tests.

**Related issues or PRs**
Please link any related GitHub [issues](https://github.com/cpacker/Letta/issues) or [PRs](https://github.com/cpacker/Letta/pulls).

**Is your PR over 500 lines of code?**
If so, please break up your PR into multiple smaller PRs so that we can review them quickly, or provide justification for its length.

**Additional context**
Add any other context or screenshots about the PR here.
